repose-aggregator/components/filters/header-user-filter/src/test/scala/org/openrepose/filters/headeruser/HeaderUserFilterTest.scala

```diff
@@ -0,0 +1,170 @@
+      wrappedRequest.getHeadersScala(User) should contain theSameElementsAs List("some-value" + DefaultQualty)
+      wrappedRequest.getHeadersScala(Groups) should contain theSameElementsAs List("some-header" + DefaultQualty)
+    }
+
+    it("adds both configured headers when they both exist in the request") {
+      filter.configurationUpdated(createConfig(List(ConfigHeader("berry", None), ConfigHeader("fruit", None))))
+      servletRequest.addHeader("berry", "banana")
+      servletRequest.addHeader("fruit", "lemon")
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      val wrappedRequest = getPostFilterRequest
+      wrappedRequest.getHeadersScala(User) should contain theSameElementsAs List("banana" + DefaultQualty, "lemon" + DefaultQualty)
+      wrappedRequest.getHeadersScala(Groups) should contain theSameElementsAs List("berry" + DefaultQualty, "fruit" + DefaultQualty)
+    }
+
+    it("does NOT add a configured header if it does not exist in the request") {
+      filter.configurationUpdated(createConfig(List(ConfigHeader("nope", None))))
+      servletRequest.addHeader("best", "north")
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      val wrappedRequest = getPostFilterRequest
+      wrappedRequest.getHeadersScala(User) shouldBe empty
+      wrappedRequest.getHeadersScala(Groups) shouldBe empty
+    }
+
+    it("does NOT add a configured header when its value in the request is empty") {
+      filter.configurationUpdated(createConfig(List(ConfigHeader("maybe", None))))
+      servletRequest.addHeader("maybe", "")
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      val wrappedRequest = getPostFilterRequest
+      wrappedRequest.getHeadersScala(User) shouldBe empty
+      wrappedRequest.getHeadersScala(Groups) shouldBe empty
+    }
+  }
+
+  describe("extracting the header values") {
+    it("adds the header value using the default quality when one is not configured") {
+      filter.configurationUpdated(createConfig(List(ConfigHeader("some-header", None))))
+      servletRequest.addHeader("some-header", "some-value")
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      val wrappedRequest = getPostFilterRequest
+      wrappedRequest.getHeader(User) shouldBe "some-value" + DefaultQualty
+    }
+
+    it("adds the first header value only") {
+      filter.configurationUpdated(createConfig(List(ConfigHeader("some-header", None))))
+      servletRequest.addHeader("some-header", "some-value,another-value,yet-another-value,omg,bbq")
+      servletRequest.addHeader("some-header", "why-not-one-more")
+
+      filter.doFilter(servletRequest, servletResponse, filterChain)
+
+      val wrappedRequest = getPostFilterRequest
+      wrappedRequest.getHeader(User) shouldBe "some-value" + DefaultQualty
+    }
+  }
+
+  private def getPostFilterRequest = filterChain.getRequest.asInstanceOf[HttpServletRequestWrapper]
```

Comment:
Definitely.  I didn't have to verify anything in the response, so I went this route here.